### PR TITLE
defer child component update so we create fewer circular dependnecies

### PR DESCRIPTION
**Pull request by boourns**
@lemonmade @eapache please review

so here's the situation: right now it's unintentionally too easy to trigger the "action dispatched while dispatching another action".  Here's the flow in question:

- we render the application, SessionStore is still loading from disk.
- Load from disk finishes which dispatches a SessionStore event
- this re-renders the application component, which now decides to mount the `Login` component.
- `Login.componentWillMount` fires `FormStore.updateForm({initial values})`.

so IMO the correct solution when using our React container, is that when we are woken up due to the store updating, to delegate the call to our listening store out of this scope so that it's free to fire events.

### CODE REVIEW

File: src/createConnect.js [hunk 1]
```diff
@@ -25,7 +25,9 @@ function createConnect(React: Object): Function {
         },
 
         render() {
-          return React.createElement(component, this.props);
+          setImmediate(() => {
+            return React.createElement(component, this.props);
```
Comment (eapache) [inline src/createConnect.js]: this won't actually render anything, since the outer render method is now returning `undefined`

Comment (boourns) [inline src/createConnect.js]: woops, im dumb. i meant to write the child `forceUpdate`.  copy-pasta slip

Comment (boourns) [inline src/createConnect.js]: @eapache fixed and rebased now

**Comment by boourns** [comment]
this doesn't work for other reasons, it breaks all our Fields because now the new text character shows up too late when you type.

**Pull request closed without merging**